modules/Chat/views/Detail.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * Chat Detail View Class.
+ *
+ * @package   View
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Mariusz Krzaczkowski <m.krzaczkowski@yetiforce.com>
+ * @author    Arkadiusz Adach <a.adach@yetiforce.com>
+ * @author    Tomasz Poradzewski <a.adach@yetiforce.com>
+ */
+class Chat_Detail_View extends \App\Controller\View
+{
+	use \App\Controller\ExposeMethod;
+
+	/**
+	 * Record model.
+	 *
+	 * @var \Vtiger_Record_Model
+	 */
+	private $recordModel;
+
+	/**
+	 * Constructor with a list of allowed methods.
+	 */
+	public function __construct()
+	{
+		parent::__construct();
+		$this->exposeMethod('showChatTab');
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function checkPermission(App\Request $request)
+	{
+		$currentUserPriviligesModel = Users_Privileges_Model::getCurrentUserPrivilegesModel();
```

Comment:
if (\App\User::getCurrentUserId() !== \App\User::getCurrentUserRealId()) {
			throw new \App\Exceptions\NoPermitted('ERR_NOT_ACCESSIBLE', 406);
		}
if (!$request->has('recordId') || Vtiger_Record_Model::getInstanceById($request->getInteger('recordId'))) {
			$this->recordModel = Vtiger_Record_Model::getInstanceById($request->getInteger('recordId')->isViewable());

				throw new \App\Exceptions\NoPermittedToRecord('ERR_NO_PERMISSIONS_FOR_THE_RECORD', 
		}